packages/core-ethereum/src/indexer/index.ts

```diff
@@ -86,9 +87,30 @@ class Indexer extends EventEmitter {
     this.chain.subscribeBlock((b) => {
       this.onNewBlock(b)
     })
+
     this.chain.subscribeError((error: any) => {
       log(chalk.red(`etherjs error: ${error}`))
-      this.restart()
+      // if provider connection issue
+      if (
+        [error?.code, String(error)].includes(errors.SERVER_ERROR) ||
+        [error?.code, String(error)].includes(errors.TIMEOUT) ||
+        [error?.code, String(error)].includes('ECONNRESET')
+      ) {
+        log(chalk.blue('code error falls here', this.chain.getAllPendingTransactionRequests().length))
+        if (this.chain.getAllPendingTransactionRequests().length > 0) {
```

Comment:
Why is this starting multiple requests when only one failed?

Reply:
When it's failed with provider error, all the pending transactions are affected (as they cannot be set to `confirmed`), so here I wanted to resend all the pending tx. However, it shouldn't work, because the tx manager checks duplicates... Meaning even the tx that does not reach the provider cannot be resent with the exact same payload... Something is wrong here.

Reply:
Updated with some major changes in `sendTransaction`: Now `core-ethereum` uses a three-step populate-sign-broadcast approach to send transactions. This approach is the same as what's used in `ethersjs`. The reason why we use those underlying function is that it segments the "signing" and "broadcasting" parts, where transaction manager can be properly informed at the end of each part. 

- After signing, a tx is `queuing`.
- After being broadcasted, a tx is `pending`.
- After being included in a block, a tx is `mined`.
- After passing the finality threshold, a tx is `confirmed`.

With that, when the provider encounters an issue, tx-manager can identify `queuing` transactions and try to resend them.